paymentsheet/src/main/java/com/stripe/android/paymentsheet/ExternalPaymentMethodActivity.kt

```diff
@@ -0,0 +1,65 @@
+package com.stripe.android.paymentsheet
+
+import android.annotation.SuppressLint
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.stripe.android.model.PaymentMethod
+
+/**
+ * Handles external payment methods.
+ *
+ * This will call the merchant's [ExternalPaymentMethodConfirmHandler.confirmExternalPaymentMethod] when initially
+ * created. When the merchant's implementation calls [ExternalPaymentMethodResultHandler.onExternalPaymentMethodResult],
+ * it will re-start this activity with the result. This class then finishes and the class that launched this activity
+ * is responsible for responding to that result.
+ */
+internal class ExternalPaymentMethodActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val type = intent.getStringExtra(EXTRA_EXTERNAL_PAYMENT_METHOD_TYPE)
+
+        @Suppress("DEPRECATION")
+        val billingDetails = intent.getParcelableExtra<PaymentMethod.BillingDetails>(EXTRA_BILLING_DETAILS)
```

Comment:
Definitely not for this PR, but it would be nice to have a stripe extension method for this so we could use the new variant on new API versions, and suppress this warning in one place.

Reply:
good idea! https://jira.corp.stripe.com/browse/MOBILESDK-2055